src/dynsight/_internal/vision/video_to_frame.py

```diff
@@ -8,99 +8,79 @@
 if TYPE_CHECKING:
     import pathlib
 
-    import numpy as np
-
 
 @dataclass
 class Video:
-    """Load a video file and provides utilities.
+    """Load a video file and provide utilities.
+
+    This class loads a video from a file path and provides methods
+    to retrieve video information and extract frames.
+
+    * Author: Simone Martino
+
+    .. caution::
+        This part of the code is still under development and may
+        contain errors.
+
+    Parameters:
+        video_path: Path to the video file.
 
-    Attributes:
-        video_path:
-            File path to the video.
     """
 
     video_path: pathlib.Path
-    frames: list[np.ndarray] = field(default_factory=list)
+    """Path to the video file."""
+    _capture: cv2.VideoCapture = field(init=False, repr=False)
+    """OpenCV video capture object."""
+
+    def __post_init__(self) -> None:
+        """Load the the video."""
+        self._capture = cv2.VideoCapture(str(self.video_path))
```

Comment:
maybe just set the parameters you need below here, then close, so it does not stay in memory? No harm in precomputing those attributes?

Reply:
Except maybe frames, although it does seem like you are not using the capture "existing" to avoid rerunning things - maybe we can discusss this further in future?